packages/site-settings/src/setup/SetupCards.tsx

```diff
@@ -1,34 +1,51 @@
 import { Box, BoxProps, styled } from '@mui/material';
 import { GoogleAnalytics, SiteDomain, SiteVisibility } from '@site/setup/cards';
-import { CARDS } from '@site/constants';
-import { SetupCardAccordion, SetupCardContent } from '@moderntribe/wme-ui';
+import { Loader, SetupCardAccordion, SetupCardContent } from '@moderntribe/wme-ui';
 import { SetupCardTasks } from '@site/setup';
+import { useSiteSettings } from '@site/hooks';
 
 const Container = styled(Box)<BoxProps>(({ theme }) => ({
 	maxWidth: theme.spacing(100),
 	margin: `${ theme.spacing(5) } auto`,
+	'& .MuiChip-label': {
+		display: 'inline-flex',
+	}
+}));
+
+// Create styled loader component that has 10px width and height
+const StyledLoader = styled(Box)(() => ({
+	display: 'inline-flex',
+	alignItems: 'center',
+	marginRight: '4px',
+	height: '18px',
+	'& svg': {
+		fontSize: '16px',
+	}
 }));
 
 const SetupCards = () => {
+	const { siteSettingsState, isLoading } = useSiteSettings();
+	const { cards, siteVisibilityValues } = siteSettingsState;
+
 	return (
 		<Container>
 			{
-				CARDS.map((card: SetupCardAccordionInterface) => {
+				cards?.map((card: SetupCardAccordionInterface) => {
 					return (
 						<SetupCardAccordion
 							key={ card.id }
 							id={ card.id }
 							header={ card.title }
 							subHeader={ card.intro }
-							chipText={ card.chipText }
-							chipBackground={ card.chipBackground }
+							chipText={ card.id === 'site-visibility' ? <>{ isLoading ? <StyledLoader><Loader /></StyledLoader> : '' } { siteVisibilityValues?.chipText }</> : card.chipText }
```

Comment:
This conditionals to show the loader should probably be in its own component since it's got two inline "if" statements. But this will work, so I've approved it.